app/src/main/cpp/BrowserWorld.cpp

```diff
@@ -766,32 +794,34 @@ BrowserWorld::AddWidget(const WidgetPlacement& placement, int32_t aCallbackId) {
   parent->GetSurfaceTextureSize(parentWidth, parentHeight);
   parent->GetWorldSize(parentWorldWith, parentWorldHeight);
 
-  float worldWidth = placement.width * kWorldDPIRatio;
-  float worldHeight;
-
-  WidgetPtr widget = Widget::Create(m.contextWeak,
-                                    placement.widgetType,
-                                    (int32_t)(placement.width * m.displayDensity),
-                                    (int32_t)(placement.height * m.displayDensity),
-                                    worldWidth);
-  widget->SetAddCallbackId(aCallbackId);
+  float worldWidth, worldHeight;
   widget->GetWorldSize(worldWidth, worldHeight);
 
-  vrb::Vector translation = vrb::Vector(placement.translation.x() * kWorldDPIRatio,
-                                        placement.translation.y() * kWorldDPIRatio,
-                                        placement.translation.z() * kWorldDPIRatio);
+  vrb::Matrix transform = vrb::Matrix::Identity();
+  if (aPlacement.rotation != 0.0) {
+    transform = vrb::Matrix::Rotation(aPlacement.rotationAxis, aPlacement.rotation);
+  }
+
+  vrb::Vector translation = vrb::Vector(aPlacement.translation.x() * kWorldDPIRatio,
+                                        aPlacement.translation.y() * kWorldDPIRatio,
+                                        aPlacement.translation.z() * kWorldDPIRatio);
   // Widget anchor point
-  translation -= vrb::Vector((placement.anchor.x() - 0.5f) * worldWidth,
-                             placement.anchor.y() * worldHeight,
+  translation -= vrb::Vector((aPlacement.anchor.x() - 0.5f) * worldWidth,
+                             aPlacement.anchor.y() * worldHeight,
                              0.0f);
   // Parent anchor point
-  translation += vrb::Vector(parentWorldWith * placement.parentAnchor.x() - parentWorldWith * 0.5f,
-                             parentWorldHeight * placement.parentAnchor.y(),
+  translation += vrb::Vector(parentWorldWith * aPlacement.parentAnchor.x() - parentWorldWith * 0.5f,
+                             parentWorldHeight * aPlacement.parentAnchor.y(),
                              0.0f);
 
-  widget->SetTransform(parent->GetTransform().PostMultiply(vrb::Matrix::Translation(translation)));
-  m.root->AddNode(widget->GetRoot());
-  m.widgets.push_back(widget);
+  transform.TranslateInPlace(translation);
+  widget->SetTransform(parent->GetTransform().PostMultiply(transform));
+  // Fixme: Remove this once we have proper scaling of the pointer
```

Comment:
You can fix this in a follow up by setting the pointer geometry to an empty group or something.